apps/backoffice-v2/src/pages/Entities/components/CaseGeneration/context/case-generation-context/hooks/useCaseGenerationState/useCaseGenerationState.ts

```diff
@@ -0,0 +1,39 @@
+import { initialCaseGenerationContext } from '@/pages/Entities/components/CaseGeneration/context/case-generation-context/case-generation.context';
+import {
+  CaseGenerationApi,
+  CaseGenerationContext,
+} from '@/pages/Entities/components/CaseGeneration/context/case-generation-context/types';
+import { useCallback, useMemo, useState } from 'react';
+
+export const useCaseGenerationAPI = () => {
+  const [state, setState] = useState<
+    Pick<CaseGenerationContext, 'isMultipleCasesCreation' | 'isOpen'>
+  >({
+    isMultipleCasesCreation: initialCaseGenerationContext.isMultipleCasesCreation,
+    isOpen: initialCaseGenerationContext.isOpen,
+  });
+
+  const toggleMultiCaseCreation = useCallback(() => {
+    setState(prev => ({ ...prev, isMultipleCasesCreation: !prev.isMultipleCasesCreation }));
+  }, []);
+
+  const setOpen = useCallback((open: boolean) => {
```

Comment:
Then there won't be a need for `isOpen: open` either.
```suggestion
  const setOpen = useCallback((isOpen: boolean) => {
```